CanHazFunny/CanHazFunny/JokeService.cs

```diff
@@ -1,15 +1,20 @@
 ﻿using System.Net.Http;
+using Newtonsoft.Json;
 
 namespace CanHazFunny
 {
-    public class JokeService
+    public class JokeService : IGetJoke
     {
         private HttpClient HttpClient { get; } = new();
 
-        public string GetJoke()
+        public string? Joke { get; set; }
+
+        public virtual string GetJoke()
         {
-            string joke = HttpClient.GetStringAsync("https://geek-jokes.sameerkumar.website/api").Result;
-            return joke;
+            Joke = HttpClient.GetStringAsync("https://geek-jokes.sameerkumar.website/api?format=json").Result;
+            dynamic? obj = JsonConvert.DeserializeObject(Joke);
+            string json = obj!.joke;
```

Comment:
Using `dynamic?` tells the compiler that `JsonConvert.DeserializeObject(Joke)` might return a `null` reference, but then on line 16 you are referencing `obj!.joke` telling the compiler you are sure `obj` is not null. How do you know `obj` is not null if you haven't checked it? What if the user's internet connection dropped before making the API request?
```suggestion
            dynamic? obj = JsonConvert.DeserializeObject(Joke);
            
            if (obj is null) 
            {
               // Either try again, or let the user know a problem occurred.
            }
            string json = obj!.joke;
```